src/codecs/cats/cats.c

```diff
@@ -0,0 +1,34 @@
+#include <stdint.h>
+
+#include "cats.h"
+#include "crc.h"
+#include "whiten.h"
+#include "ldpc.h"
+#include "interleaver.h"
+
+cats_packet cats_create(uint8_t *data)
+{
+    cats_packet c = {.data = data, .len = 0};
```

Comment:
What is `len` used for?

Reply:
Ok, yes, it's used later on, I just didn't immediately see it. I'm just wondering if there could be ways to make sure the packet will never grow outside of boundaries of the allocated `data` length.

Reply:
What do we do for the APRS type?

The only way this would be able to grow too long is from the user specifying a very long message. We could check for this and throw a compile-time error, or we could do a bounds check at run-time and skip the CATS transmission if we'd overrun the buffer

Reply:
Yeah, let's leave this maybe as a future enhancement. We'll have to think about a good approach.